checkov/terraform/graph_builder/variable_rendering/renderer.py

```diff
@@ -366,6 +381,13 @@ def evaluate_non_rendered_values(self) -> None:
                     changed_attributes[attribute] = evaluated
             self.local_graph.update_vertex_config(vertex, changed_attributes)
 
+    @staticmethod
+    def extract_dynamic_value_in_map(dynamic_value: str) -> str:
+        dynamic_value_in_map = dynamic_value.split('.')[-1]
+        if '[' not in dynamic_value_in_map:
+            return dynamic_value_in_map
+        return dynamic_value_in_map.split('["')[-1].replace('"]', '')
```

Comment:
Can we extract this operator into a named method? The usage is not clear.
Also, please test this method